contrib/cirrus/win-sess-launch.ps1

```diff
@@ -2,24 +2,74 @@
 # tunnels the output such that WSL operations will complete
 $ErrorActionPreference = 'Stop'
 
+$syms =  [char[]]([char]'a'..[char]'z') +
+        [char[]]([char]'A'..[char]'Z') +
+        [char[]]([char]'0'..[char]'9')
+
 if ($Args.Length -lt 1) {
     Write-Object "Usage: " + $MyInvocation.MyCommand.Name + " <script>"
     Exit 1;
 }
 
-function RegenPassword {
-    param($username)
-    $syms = [char[]]([char]'a'..[char]'z' `
-               + [char]'A'..[char]'Z'  `
-               + [char]'0'..[char]'9')
+
+# Shuffle a 32 character password from $syms
+function GenerateRandomPassword {
+    param(
+        [int]$length = 32,
+        [array]$symbols = $syms
+    )
+
     $rnd = [byte[]]::new(32)
     [System.Security.Cryptography.RandomNumberGenerator]::create().getBytes($rnd)
-    $password = ($rnd | % { $syms[$_ % $syms.length] }) -join ''
-    $encPass = ConvertTo-SecureString $password -AsPlainText -Force
-    Set-LocalUser -Name $username -Password $encPass
+    $password = ($rnd | % { $symbols[$_ % $symbols.length] }) -join ''
+
     return $password
 }
 
+function IsPasswordValid {
```

Comment:
theoretically, it could fail if only the lowercase letters are picked randomly. (wild guess?)

But thinking about your comments, this function also seemed problematic, so I just dropped it and instead used the trycatch block for "validation".

Thanks for the help @l0rd